job-service/src/main/java/com/hpe/caf/services/job/api/filter/CustomRsqlVisitor.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2016-2020 Micro Focus or one of its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hpe.caf.services.job.api.filter;
+
+import com.healthmarketscience.sqlbuilder.Condition;
+import cz.jirutka.rsql.parser.ast.AndNode;
+import cz.jirutka.rsql.parser.ast.ComparisonNode;
+import cz.jirutka.rsql.parser.ast.OrNode;
+import cz.jirutka.rsql.parser.ast.RSQLVisitor;
+
+public final class CustomRsqlVisitor<T> implements RSQLVisitor<Condition, Void>

```

Comment:
Perhaps I'm missing something but I can't see the point of the split between the three classes in this package.  Personally I think I would have just had one and called it something like `RsqlToSqlConverter`.  And why is it generic - the type parameter isn't being used!

```suggestion
public final class RsqlToSqlConverter implements RSQLVisitor<Condition, Void>
```